programs/stake/src/stake_instruction.rs

```diff
@@ -6195,26 +6197,23 @@ mod tests {
                 Err(InstructionError::NotEnoughAccountKeys),
             ),
         ] {
+            let mut feature_set = FeatureSet::all_enabled();
+            if !is_feature_enabled {
+                feature_set.deactivate(
+                    &feature_set::add_get_minimum_delegation_instruction_to_stake_program::id(),
+                );
+            }
+
             mock_process_instruction(
                 &id(),
                 Vec::new(),
                 &instruction.data,
                 transaction_accounts.clone(),
                 instruction_accounts.clone(),
                 None,
+                Some(Arc::new(feature_set)),
                 expected_result,
-                if is_feature_enabled {
-                    |first_instruction_account, invoke_context| {
-                        super::process_instruction(first_instruction_account, invoke_context)
-                    }
-                } else {
-                    |first_instruction_account, invoke_context| {
-                        let mut feature_set = FeatureSet::all_enabled();
-                        feature_set.deactivate(&feature_set::add_get_minimum_delegation_instruction_to_stake_program::id());
-                        invoke_context.feature_set = Arc::new(feature_set);
-                        super::process_instruction(first_instruction_account, invoke_context)
-                    }
-                },
```

Comment:
By adding the feature set override, I can remove all this and accomplish the same goal much more clearly.